coriolis/osmorphing/osmount/base.py

```diff
@@ -298,6 +299,99 @@ def _get_volume_block_devices(self):
+                self._event_manager.progress_update(
+                    "Failed to mount and scan device '%s'" % dev_path)
+                LOG.warn(
+                    "Failed to mount and scan device '%s':\n%s",
+                    dev_path, utils.get_exception_details())
+                utils.ignore_exceptions(self._exec_cmd)(
+                    "sudo umount %s" % tmp_dir
+                )
+                utils.ignore_exceptions(self._exec_cmd)(
+                    "sudo rmdir %s" % tmp_dir
+                )
+                continue
+
+            LOG.debug("Contents of device %s:\n%s", dev_path, dirs)
+
+            if all_files and dirs:
+                common = [i if i in dirs else None for i in all_files]
+                if not all(common):
+                    self._exec_cmd('sudo umount %s' % tmp_dir)
+                    continue
+
+                dev_name = dev_path
+                self._exec_cmd('sudo umount %s' % tmp_dir)
+            elif one_of_files and dirs:
+                common = [i for i in one_of_files if i in dirs]
+                if len(common) > 0:
+                    dev_name = dev_path
+                    self._exec_cmd('sudo umount %s' % tmp_dir)
+                    break
+            else:
+                self._exec_cmd('sudo umount %s' % tmp_dir)
+                continue
+
+        if dev_path:
+            devices.remove(dev_path)
+        return dev_name
+
+    def _find_and_mount_root(self, devices):
+        files = ["etc", "bin", "sbin", "boot"]
+        os_root_dir = None
+        os_root_device = self._find_dev_with_contents(
+            devices, all_files=files,
+        )
+
+        if os_root_device is None:
+            raise exception.OperatingSystemNotFound(
+                "root partition not found")
+
+        try:
+            tmp_dir = self._exec_cmd('mktemp -d').decode().split('\n')[0]
+            self._exec_cmd('sudo mount %s %s' % (os_root_device, tmp_dir))
+            os_root_dir = tmp_dir
+        except Exception:
+            self._event_manager.progress_update(
+                "Failed to mount root device '%s'" % os_root_device)
+            LOG.warn(
+                "Failed to mount root device '%s':\n%s",
+                os_root_device, utils.get_exception_details())
+            utils.ignore_exceptions(self._exec_cmd)(
+                "sudo umount %s" % tmp_dir
+            )
+            utils.ignore_exceptions(self._exec_cmd)(
+                "sudo rmdir %s" % tmp_dir
+            )
+            raise
+
+        for directory in ['proc', 'sys', 'dev', 'run']:
+            mount_dir = os.path.join(os_root_dir, directory)
+            if not utils.test_ssh_path(self._ssh, mount_dir):
+                LOG.info(
+                    "No '%s' directory in mounted OS. Skipping mount.",
+                    directory)
+                continue
+            self._exec_cmd(
+                'sudo mount -o bind /%(dir)s/ %(mount_dir)s' %
```

Comment:
Any reason for the trailing slash on the first arg?

Reply:
code was moved, not changed. But as a note, the trailing slash disambiguates symlinks from actual folders.

```bash
gabriel@rossak:/tmp/test$ mkdir a_dir
gabriel@rossak:/tmp/test$ ln -s a_dir a_symlink
gabriel@rossak:/tmp/test$ ln -s a_dir2 a_broken_symlink
gabriel@rossak:/tmp/test$
gabriel@rossak:/tmp/test$ ls -l a_dir
total 0
gabriel@rossak:/tmp/test$ ls -l a_symlink
lrwxrwxrwx 1 gabriel gabriel 5 sep  3 16:18 a_symlink -> a_dir
gabriel@rossak:/tmp/test$ ls -l a_symlink/
total 0
gabriel@rossak:/tmp/test$ ls -l a_broken_symlink
lrwxrwxrwx 1 gabriel gabriel 6 sep  3 16:18 a_broken_symlink -> a_dir2
gabriel@rossak:/tmp/test$ ls -l a_broken_symlink/
ls: cannot access 'a_broken_symlink/': No such file or directory
```

Reply:
Had no clue, thanks for the clarification!